src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -267,6 +292,30 @@ protected override (bool hasNext, int nextSlot, int nextIndex) MoveNext(int prev
             }
         }
 
+        internal override (bool hasNext, int nextSlot, int nextIndex) MoveNextReversed(int previousSlot, int previousIndex)
+        {
+            switch (previousSlot)
+            {
+                case int.MaxValue:
+                    if (Initializer == null) goto case 1;
+                    else return (true, 1, 0);
+
+                case 1:
+                    if (!Arguments.IsEmpty) return (true, 0, Arguments.Length - 1);
+                    else goto case 0;
+
+                case 0 when previousIndex > 0:
+                    return (true, 0, previousIndex - 1);
+
+                case 0:
+                case -1:
+                    return (false, -1, 0);
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue((previousSlot, previousIndex));
```

Comment:
The `MoveNextReversed` implementations differ on how they handle invalid `previousSlot` values. Some throw, some return `false`. Feel like the implementations should be consistent.